terraform/dns.tf

```diff
@@ -330,7 +330,28 @@ locals {
       hostname = "wiki.nixos.org"
       type     = "AAAA"
       value    = "2a01:4f9:c012:8178::"
-    }
+    },
+
+    # Mailserver configuration for `mail-test.nixos.org`
+    {
+      hostname = "mail-test.nixos.org"
+      type     = "MX"
+      value    = "umbriel.nixos.org"
+    },
+    {
+      hostname = "wiki.nixos.org"
+      type     = "TXT"
+      value    = "v=spf1 a:mail-test.nixos.org -all"
```

Comment:
@Mic92, I see we're hardcoding IPs here: https://github.com/NixOS/infra/blob/fb5e9aeb6c1621c98d2c9ba029613f3be7c9e1db/terraform/dns.tf#L317, but these docs use a domain name: https://nixos-mailserver.readthedocs.io/en/latest/setup-guide.html#set-a-spf-record.

Any reason to do the IPs rather than a domain name?

Reply:
`a:` seems to be ipv4 only? We also have ipv6.

Reply:
Despite the name, I think it's dual ipv4/ipv6. From https://datatracker.ietf.org/doc/html/rfc7208#section-5.3:

> An address lookup is done on the <target-name> using the type of lookup (A or AAAA) appropriate for the connection type (IPv4 or IPv6)

[From experience](https://gitlab.com/simple-nixos-mailserver/nixos-mailserver/-/merge_requests/342), I know that my personal mailserver is (now) successfully sending emails to Gmail over IPv6.

Reply:
@mweinelt:

1. I like the thought of using `mx` rather than `a`.
2. I'm not sure about the `~all` vs `-all`. I'd be inclined to keep it as strict as possible until we have a good reason not to.

Reply:
Yeah, I prefer strictness as well.

Reply:
I've confirmed with my personal test mailserver that `mx` seems to work here. I've changed this SPF record to `v=spf1 mx -all`.

I'm going to mark this resolved.